instrumentor/runtimemigration/runtimemigration.go

```diff
@@ -0,0 +1,393 @@
+		if len(annotationEnvVarsForContainer) == 0 {
+			state := v1alpha1.ProcessingStateSucceeded
+			containerRuntimeDetails.RuntimeUpdateState = &state
+		}
+
+		for envKey, envValue := range annotationEnvVarsForContainer {
+
+			// The containerRuntimeDetails might already include the EnvFromContainerRuntime if the runtime inspection was executed before the migration modified the environment variables.
+			// In this case, we want to avoid overwriting the value set by Odiglet.
+			// This check is only for safety, as we have already skipped processed containers.
+			isEnvVarAlreadyExists := isEnvVarPresent(containerRuntimeDetails.EnvFromContainerRuntime, envKey)
+			if isEnvVarAlreadyExists {
+				continue
+			}
+
+			// if envValue is nil, it means that the value in the manifest is come from the runtime by the envOverwriter.
+			// In his case, we mark as succeeded and set the EnvFromContainerRuntime as clean version of the manifest env values (without odigos additions).
+			if envValue == nil {
+				containerEnvFromManifestValue := k8scontainer.GetContainerEnvVarValue(&containerObject, envKey)
+				if containerEnvFromManifestValue != nil {
+					workloadEnvVarWithoutOdigosAdditions := cleanUpManifestValueFromOdigosAdditions(envKey, *containerEnvFromManifestValue)
+
+					if workloadEnvVarWithoutOdigosAdditions != "" {
+						envVarWithoutOdigosAddition := v1alpha1.EnvVar{Name: envKey, Value: workloadEnvVarWithoutOdigosAdditions}
+						containerRuntimeDetails.EnvFromContainerRuntime = append(containerRuntimeDetails.EnvFromContainerRuntime, envVarWithoutOdigosAddition)
+						state := v1alpha1.ProcessingStateSucceeded
+						containerRuntimeDetails.RuntimeUpdateState = &state
+					}
+
+				}
+			} else {
+				// If envKey exists and != nil, it indicates that the environment variable originally came from the manifest.
+				// In this case, we will set the RuntimeUpdateState to ProcessingStateSkipped.
+				state := v1alpha1.ProcessingStateSkipped
+				containerRuntimeDetails.RuntimeUpdateState = &state
+			}
+		}
+	}
+	return nil
+}
+
+func contains(workloadNames map[string]*v1alpha1.InstrumentationConfig, workloadName string) bool {
+	_, exists := workloadNames[workloadName]
+	return exists
+}
+
+func cleanUpManifestValueFromOdigosAdditions(manifestEnvVarKey string, manifestEnvVarValue string) string {
+	_, exists := envOverwrite.EnvValuesMap[manifestEnvVarKey]
+	if exists {
+		// clean up the value from all possible odigos additions
+		for _, value := range envOverwrite.GetPossibleValuesPerEnv(manifestEnvVarKey) {
+			manifestEnvVarValue = strings.ReplaceAll(manifestEnvVarValue, value, "")
+		}
+		withoutTrailingColon := cleanTrailingChar(manifestEnvVarValue, ":")
+		withoutTrailingAndLeadingSpace := strings.TrimSpace(withoutTrailingColon)
+		return withoutTrailingAndLeadingSpace
+	} else {
+		// manifestEnvVarKey does not exist in the EnvValuesMap
+		return ""
```

Comment:
This is just for safety and should never happen, right?

Reply:
correct, it is not really needed